### PR TITLE
feat: add support for interaction events in sfmc

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## What are the changes introduced in this PR?

We are adding support for sending interaction events in SFMC

## What is the related Linear task?

Resolves INT-1425

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3109/test-report.html

File: src/v0/destinations/sfmc/transform.js [hunk 1]
```diff
@@ -182,6 +183,26 @@ const responseBuilderForInsertData = (
   return response;
 };
 
+// DOC : https://developer.salesforce.com/docs/marketing/marketing-cloud/references/mc_rest_interaction/postEvent.html
+
+const responseBuilderForMessageEvent = (message, subDomain, authToken, hashMapEventDefinition) => {
```
Comment (sanpj2292) [inline src/v0/destinations/sfmc/transform.js:188]: Are we exporting this for testing purposes ?

File: src/v0/destinations/sfmc/transform.js [hunk 2]
```diff
@@ -234,10 +258,12 @@ const responseBuilderSimple = async (message, category, destination) => {
     if (typeof message.event !== 'string') {
       throw new ConfigurationError('Event name must be a string');
     }
+    if (hashMapEventDefinition[message.event.toLowerCase()]) {
+      return responseBuilderForMessageEvent(message, subDomain, authToken, hashMapEventDefinition);
```
Comment (sanpj2292) [inline src/v0/destinations/sfmc/transform.js:262]: Should we add a component test to cover our changes ?

File: src/v0/destinations/sfmc/transform.js (hunk 1 above)
Comment (yashasvibajpai) [reply src/v0/destinations/sfmc/transform.js:188]: Yes, for `transform.test.js`

File: src/v0/destinations/sfmc/transform.js (hunk 2 above)
Comment (yashasvibajpai) [reply src/v0/destinations/sfmc/transform.js:262]: Addressed.

**Review by sanpj2292 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**